gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -146,13 +144,12 @@ private void publishFileSet(CopyableFile.DatasetAndPartition datasetAndPartition
 
     CopyableDatasetMetadata metadata = CopyableDatasetMetadata.deserialize(
         datasetWorkUnitStates.iterator().next().getProp(CopySource.SERIALIZED_COPYABLE_DATASET));
-    Path datasetWriterOutputPath = new Path(new Path(writerOutputDir, datasetAndPartition.identifier()),
-        PathUtils.withoutLeadingSeparator(metadata.getDatasetTargetRoot()));
+    Path datasetWriterOutputPath = new Path(this.writerOutputDir, datasetAndPartition.identifier());
 
     log.info(String
         .format("Publishing fileSet from %s to %s", datasetWriterOutputPath, metadata.getDatasetTargetRoot()));
 
-    HadoopUtils.renameRecursively(fs, datasetWriterOutputPath, metadata.getDatasetTargetRoot());
+    HadoopUtils.renameRecursively(fs, datasetWriterOutputPath, findPathRoot(metadata.getDatasetTargetRoot()));
```

Comment:
It doesn't, line 147 was changed as well so the publish behavior is similar. The directory structure in dataset output mimics the directory structure in the file system. Previously we would move at the dataset level, so if dataset is /foo/bar/dataset, but /foo/bar didn't exist, it would be created automatically with default permissions (because we attempt to move .../staging/foo/bar/dataset to /foo/bar/dataset). 
Now we are moving at the highest level possible, i.e. highest level directory that doesn't exist in the target (we will move /staging/foo/bar to /foo/bar). This has the effect of creating any missing directories with the appropriate permissions.